argilla/src/argilla/settings/_templates/_features.py

```diff
@@ -0,0 +1,166 @@
+                        }
+                    ),
+                    "rejected": Sequence(
+                        feature={
+                            "content": Value(dtype="string"),
+                            "role": Value(dtype="string"),
+                        }
+                    ),
+                }
+            ),
+            Features(
+                {
+                    "instruction": Value("string"),
+                    "input": Value("string"),
+                    "response1": Value("string"),
+                    "response2": Value("string"),
+                }
+            ),
+            Features(
+                {
+                    "question": Value("string"),
+                    "answer1": Value("string"),
+                    "answer2": Value("string"),
+                }
+            ),
+        ]
+
+    @staticmethod
+    def response_rating():
+        return [
+            Features(
+                {
+                    "instruction": Value("string"),
+                    "input": Value("string"),
+                    "output": Value("string"),
+                }
+            ),
+            Features(
+                {
+                    "prompt": Value("string"),
+                    "response": Value("string"),
+                }
+            ),
+            Features(
+                {
+                    "question": Value("string"),
+                    "answer": Value("string"),
+                }
+            ),
+        ]
+
+    @staticmethod
+    def _compare_features(dataset_features, predefined_features):
+        if not set(dataset_features.keys()).issuperset(set(predefined_features.keys())):
+            return False
+
+        dataset_features = {k: v for k, v in dataset_features.items() if k in predefined_features}
+        return dataset_features == predefined_features
+
+    @classmethod
+    def _get_preset(cls, dataset: Dataset, labels: Optional[list[str]] = None):
+        if labels is not None and any(
+            cls._compare_features(dataset.features, features) for features in cls.document_classification(labels=labels)
+        ):
+            return FeatureType.document_classification
+        elif any(cls._compare_features(dataset.features, features) for features in cls.response_ranking()):
+            return FeatureType.response_ranking
+        elif any(cls._compare_features(dataset.features, features) for features in cls.response_rating()):
+            return FeatureType.response_rating
+        else:
+            return None
+
+    @classmethod
+    def from_dataset(cls, dataset: Dataset, labels: Optional[list[str]] = None):
+        feature_preset = cls._get_preset(dataset, labels=labels)
+        if feature_preset is None:
+            raise ValueError("The dataset features do not match any of the feature presets.")
+        elif feature_preset == FeatureType.document_classification:
+            return cls.for_document_classification(labels=labels)
```

Comment:
should we consider just using the unique values from the dataset, when no labels are passed?

Reply:
Yeah. This would make it more adaptable. We would need to deal with labels as `Value` or `ClassLabel` and the latter's name, so there would be some validation logic.